File: day1.py
#imports
import os

# set global path
path = os.path.dirname(__file__)

#main
def part1():
    global path
    with open(f'{path}\day1.txt', 'r') as file:
        last_input = float("inf")
        increased = 0
        decreased = 0
        for item in file:
            item = int(item.strip())
            if item > last_input:
                increased += 1
            else:
                decreased += 1
            last_input = item
        print(f"\nHow many measurements are larger than the previous measurement?\n{increased}")

def part2():
    global path
    with open(f'{path}\day1.txt', 'r') as file:
        last_input = float("inf")
        increased = 0
        decreased = 0
        items = []
        for item in file:
            item = int(item.strip())
            items.append(item)
        for i in range(len(items) - 2):
            value = sum(items[i:i+3])
            if value > last_input:
                increased += 1
            else:
                decreased += 1
            last_input = value
        print(f"\nHow many sums are larger than the previous sum?\n{increased}")

def main():
    part1()
    part2()

if __name__ == "__main__":
    main()

File: C#/AOC2021/days/day1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace ConsoleApp1
{
    static class day1
    {
        static void Main(string[] args)
        {
            Part1();
        }

        static void Part1()
        {
            //How many measurements are larger than the previous measurement?

            string[] inputs = File.ReadAllLines("C:\\Users\\User\\Documents\\DriveSync\\GitHub\\AOC2021\\C#\\AOC2021\\inputs\\day1.txt", Encoding.UTF8);
            float last_input = float.PositiveInfinity;
            int increased = 0;
            int decreased = 0;

            for (int i = 0; i < inputs.Length; i++)
            {
                if (int.Parse(inputs[i]) > last_input)
                {
                    increased += 1;
                }
                else
                {
                    decreased += 1;
                }
                last_input = int.Parse(inputs[i]);
            }

            Console.WriteLine(String.Format("How many measurements are larger than the previous measurement?\n{0}", increased));
        }

        static void Part2()
        {
            //How many sums are larger than the previous sum?
        }

    }
}